plugins/module_utils/network/sonic/config/dhcp_snooping/dhcp_snooping.py

```diff
@@ -0,0 +1,636 @@
+                        'iptype': 'ipv' + str(v),
+                        'config': {
+                            'mac': entry.get('mac_addr'),
+                            'iptype': 'ipv' + str(v),
+                            'vlan': str(entry.get('vlan_id')),
+                            'interface': entry.get('intf_name'),
+                            'ip': entry.get('ip_addr'),
+                        }
+                    })
+            # as long as at least one source binding has mac address, we have data. should happen since it is a required field
+            payload = {'openconfig-dhcp-snooping:entry': entries}
+            uri = self.binding_uri
+            requests.append({'path': uri, 'method': PATCH, 'data': payload})
+
+        return requests
+
+    def get_delete_all_have_requests(self, afis):
+        '''creates and builds list of requests to delete all current dhcp snooping config for ipv4 and ipv6'''
+        requests = []
+        requests.extend(self.get_delete_all_afi_requests(afis.get('have_ipv4')))
+        requests.extend(self.get_delete_all_afi_requests(afis.get('have_ipv6')))
+
+        return requests
+
+    def get_delete_specific_requests(self, afis):
+        '''creates and returns list of requests to delete afi settings.
+           Checks if clearing settings for a ip family or just matching fields in config'''
+        modified_afi_commands = []
+        requests = []
+
+        want_ipv4 = afis.get('want_ipv4')
+        want_ipv6 = afis.get('want_ipv6')
+        have_ipv4 = afis.get('have_ipv4')
+        have_ipv6 = afis.get('have_ipv6')
+
+        if want_ipv4:
+            if want_ipv4.keys() == set(["afi", "enabled", "verify_mac"])\
+                    and self.defaults_unchanged(want_ipv4):
+                # AFI dict specified in playbook looks like it's default, taking this to mean want to clear all settings for AFI
+                modified_afi_commands.append(deepcopy(have_ipv4))
+                requests.extend(self.get_delete_all_afi_requests(have_ipv4))
+            else:
+                ipv4_commands, ipv4_requests = self.get_delete_specific_afi_fields_requests(want_ipv4, have_ipv4)
+                requests.extend(ipv4_requests)
+                if ipv4_commands:
+                    ipv4_commands["afi"] = deepcopy(want_ipv4["afi"])
+                    modified_afi_commands.append(ipv4_commands)
+        if want_ipv6:
+            if want_ipv6.keys() == set(["afi", "enabled", "verify_mac"])\
+                    and self.defaults_unchanged(want_ipv6):
+                modified_afi_commands.append(deepcopy(have_ipv6))
+                requests.extend(self.get_delete_all_afi_requests(have_ipv6))
+            else:
+                ipv6_commands, ipv6_requests = self.get_delete_specific_afi_fields_requests(want_ipv6, have_ipv6)
+                requests.extend(ipv6_requests)
+                if ipv6_commands:
+                    ipv6_commands["afi"] = deepcopy(want_ipv6["afi"])
```

Comment:
Since `afi` is a str option, deepcopy is not required.
```suggestion
                    ipv6_commands["afi"] = want_ipv6["afi"]
```